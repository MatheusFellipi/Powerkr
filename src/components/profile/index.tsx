import { Avatar, Text, View } from "tamagui";

export default function Home() {
  return (
    <View>
      <Avatar
        circular
        size="$15"
      >
        <Avatar.Image
          accessibilityLabel="Cam"
          src="https://images.unsplash.com/photo-1548142813-c348350df52b?&w=150&h=150&dpr=2&q=80"
        />
        <Avatar.Fallback backgroundColor="$blue10" />
      </Avatar>
      <Text>Colaborador nome</Text>
      <Text>Departamento</Text>
    </View>
  );
}
