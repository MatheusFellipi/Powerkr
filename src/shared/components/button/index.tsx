import { Button, styled } from "tamagui";

export const BtnComponent = styled(Button, {
  fontSize: 14,
  borderRadius: 24,
  fontFamily: "InterBold",
  borderColor: "#4B5565",
  bg: "#202939",
  variants: {
    tabs: {
      activated: {},
      disabled: {
        borderColor: "$colorTransparent",
        bg: "$colorTransparent"
      }
    },
    styles: {
      login: {
        bg: "#F7B519",
        color: "#000"
      }
    }
  } as const
});
