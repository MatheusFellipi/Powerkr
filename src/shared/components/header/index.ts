import { Container } from "@shared/components/header/container";
import { Icon } from "@shared/components/header/icon";
import { TextComponent } from "@shared/components/textComponent";

export const HeaderComponent = {
  Text: TextComponent,
  Container: Container,
  Icon: Icon
};
