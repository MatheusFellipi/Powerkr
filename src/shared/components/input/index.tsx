import { useState } from "react";
import { Pressable, TextInputProps } from "react-native";
import { Icons } from "@assets/index";
import { Input, InputProps, Label, View } from "tamagui";

type Props = { label: string } & InputProps;

export const InputComponent = ({
  label,
  id,
  onChangeText,
  secureTextEntry,
  ...props
}: Props) => {
  const [showPassword, setShowPassword] = useState(secureTextEntry);
  return (
    <View>
      <Label
        color={"#CDD5DF"}
        htmlFor={id}
      >
        {label}
      </Label>
      <View
        fd="row"
        jc="space-between"
        ai="center"
        borderRadius={24}
        bg="#364152"
        borderWidth={1}
        borderColor="#4B5565"
      >
        <Input
          {...props}
          width={"90%"}
          borderRadius={24}
          secureTextEntry={showPassword}
          bg="$backgroundTransparent"
          borderColor="$backgroundTransparent"
        />
        {secureTextEntry && (
          <Pressable
            onPress={() => setShowPassword(!showPassword)}
            style={{
              width: "10%",
              paddingRight: 10
            }}
          >
            {showPassword ? (
              <Icons.EyeOn
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            ) : (
              <Icons.EyeOff
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            )}
          </Pressable>
        )}
      </View>
    </View>
  );
};
