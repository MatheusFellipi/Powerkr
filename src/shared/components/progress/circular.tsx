import { AnimatedCircularProgress } from "react-native-circular-progress";
import { TextComponent } from "@shared/components/textComponent";
import { View } from "tamagui";

export const CircularProgressComponent = ({
  description,
  value = 0
}: ProgressCircularProps) => (
  <AnimatedCircularProgress
    size={134}
    width={10}
    backgroundWidth={10}
    fill={value}
    tintColor="#44CB4B"
    backgroundColor="#3d5875"
    arcSweepAngle={240}
    rotation={240}
    lineCap="round"
  >
    {() => (
      <Legend
        description={description}
        value={value}
      />
    )}
  </AnimatedCircularProgress>
);

const Legend = ({ value, description }: ProgressCircularProps) => (
  <View ai="center">
    <TextComponent>{value}</TextComponent>
    <TextComponent
      pin="description"
      textAlign="center"
    >
      {description}
    </TextComponent>
  </View>
);
