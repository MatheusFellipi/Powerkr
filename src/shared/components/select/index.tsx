import { useMemo, useState } from "react";
import { Icons } from "@assets/index";
import { Adapt, Select as S, Sheet } from "tamagui";

const items = [{ name: "Todos" }];

export function SelectComponent() {
  const [val, setVal] = useState("todos");
  return (
    <S
      id="food"
      value={val}
      onValueChange={setVal}
      disablePreventBodyScroll
    >
      <S.Trigger
        borderRadius={25}
        width={131}
        bg="$colorTransparent"
        borderColor="#4B5565"
        iconAfter={Icons.ArrowDown}
      >
        <S.Value placeholder="Something" />
      </S.Trigger>
      <Adapt
        when="sm"
        platform="touch"
      >
        <Sheet
          modal
          dismissOnSnapToBottom
          animationConfig={{
            type: "spring",
            damping: 20,
            mass: 1.2,
            stiffness: 250
          }}
        >
          <Sheet.Frame>
            <Sheet.ScrollView>
              <Adapt.Contents />
            </Sheet.ScrollView>
          </Sheet.Frame>
          <Sheet.Overlay
            animation="lazy"
            enterStyle={{ opacity: 0 }}
            exitStyle={{ opacity: 0 }}
          />
        </Sheet>
      </Adapt>
      <S.Content>
        <S.Viewport>
          <S.Group>
            {useMemo(
              () =>
                items.map((item, i) => {
                  return (
                    <S.Item
                      index={i}
                      key={item.name}
                      value={item.name.toLowerCase()}
                    >
                      <S.ItemText>{item.name}</S.ItemText>
                    </S.Item>
                  );
                }),

              [items]
            )}
          </S.Group>
        </S.Viewport>
      </S.Content>
    </S>
  );
}
