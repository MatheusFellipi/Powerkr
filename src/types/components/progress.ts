type ProgressCircularProps = {
  value: number;
  description: string;
};
