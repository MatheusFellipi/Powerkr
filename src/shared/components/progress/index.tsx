import { CircularProgressComponent } from "@shared/components/progress/circular";

export const Progress = {
  Circular: CircularProgressComponent
};
