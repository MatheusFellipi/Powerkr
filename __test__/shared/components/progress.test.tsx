import { Progress } from "@shared/components/progress";
import { render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import config from "../../../tamagui.config";

describe("<App />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <Progress.Circular
          description="Test"
          value={1}
        />
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(screen.getByText("Test")).toBeDefined();
  });
});
