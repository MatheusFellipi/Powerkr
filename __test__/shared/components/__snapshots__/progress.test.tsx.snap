// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={{}}
>
  <RNSVGSvgView
    bbHeight="134"
    bbWidth="134"
    focusable={false}
    height={134}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 134,
          "width": 134,
        },
      ]
    }
    width={134}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        matrix={
          [
            -0.5000000000000004,
            -0.8660254037844385,
            0.8660254037844385,
            -0.5000000000000004,
            42.47629794644265,
            158.5237020535574,
          ]
        }
      >
        <RNSVGPath
          d="M 13.319414925147349 98.02248839183451 A 62 62 0 1 0 67 5"
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={
            {
              "payload": 4282210421,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeWidth="10"
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 57,
        "height": 114,
        "justifyContent": "center",
        "left": 10,
        "overflow": "hidden",
        "position": "absolute",
        "top": 10,
        "width": 114,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#fff",
            "fontFamily": "Inter",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      >
        1
      </Text>
      <Text
        fontFamily="Inter"
        style={
          {
            "color": "#9AA4B2",
            "fontFamily": "Inter",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
        suppressHighlighting={true}
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;
